### PR TITLE
Competitive programming: Greedy-Added CarSell problem

### LINKED ISSUE #7594: Competitive Programming: Greedy-Car Sell Problem
### Description

Documentation for car selling problem to achieve maximum possible profit. Code snippet added is in C++.

### Domain

Competitive Programming

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by shivam-iitk**
<hr>

## Description 📜
This is a problem using Greedy Algorithms. In this problem, we have to sell all cars in an optimal way to maximize profit.

Fixes #7594

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Competitive #Programming<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by shivam-iitk** [comment]
@atarax665 review, please.

File: Competitive_Programming/C++/Greedy/CarSell_Problem/CarSell.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+## Problem Definition
+
+MAXIMUM PROFIT:
+
+A man owns N cars (numbered from 1 to N). He wishes to sell all cars over N days by selling exactly one car per day. For each valid i, the initial price of the i-th car is Pi. Due to depreciation, the price of each car decreases by 1 unit per day until it is sold. Note that the price of a car cannot drop below 0 no matter how many days have passed, i.e. when the price of a car reaches 0 on some day then it remains 0 in all subsequent days. Find the maximum profit he can make if he sells his cars in an optimal way in order to maximise his profit. The output can be very large so compute it to modulo of 1000000007(10^9+7).
+
+
+## EXPLANATION
+We use Greedy approach for solving this problem.
```
Comment (atarax665) [inline Competitive_Programming/C++/Greedy/CarSell_Problem/CarSell.md:11]:
```suggestion
We use the greedy approach for solving this problem. 
```

File: Competitive_Programming/C++/Greedy/CarSell_Problem/CarSell.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+## Problem Definition
+
+MAXIMUM PROFIT:
+
+A man owns N cars (numbered from 1 to N). He wishes to sell all cars over N days by selling exactly one car per day. For each valid i, the initial price of the i-th car is Pi. Due to depreciation, the price of each car decreases by 1 unit per day until it is sold. Note that the price of a car cannot drop below 0 no matter how many days have passed, i.e. when the price of a car reaches 0 on some day then it remains 0 in all subsequent days. Find the maximum profit he can make if he sells his cars in an optimal way in order to maximise his profit. The output can be very large so compute it to modulo of 1000000007(10^9+7).
+
+
+## EXPLANATION
+We use Greedy approach for solving this problem. 
+As we already know that price of any given car can never become negative. So, we will start by selling the car with maximum price and then proceed to car with lower price.
```
Comment (atarax665) [inline Competitive_Programming/C++/Greedy/CarSell_Problem/CarSell.md:12]:
```suggestion
As we already know the price of any given car can never become negative. So, we will start by selling the car with the maximum price and then proceed to the car with a lower price.
```

File: Competitive_Programming/C++/Greedy/CarSell_Problem/CarSell.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	    for(i=n-1;i>=0;i--)
+	    {
+                if(v[i]-n+1+i>0)k+=v[i]-n+1+i;
+                else break;
+	    }
+	    k=k%1000000007;
+	    cout<<k<<endl;
+	}
+	return 0;
+}
```
Comment (atarax665) [inline Competitive_Programming/C++/Greedy/CarSell_Problem/CarSell.md:55]: Improve the indentation of the code.

**Comment by shivam-iitk** [comment]
please merge now @atarax665

File: Competitive_Programming/C++/Greedy/CarSell_Problem/CarSell.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	for(i=n-1;i>=0;i--)
+	{
+    	if(v[i]-n+1+i>0)k+=v[i]-n+1+i;
+        else break;
+	}
+	k=k%1000000007;
+	//printing maximum possible profit
+	cout<<k<<endl;
+	return 0;
+}
```
Comment (atarax665) [inline Competitive_Programming/C++/Greedy/CarSell_Problem/CarSell.md:55]:
```suggestion
#include <bits/stdc++.h>
using namespace std;
int main()
{
	long long int n,k,i;
	cin>>n;
	k=0;
        //create a vector to store price of cars
	vector <long long int> v(n);
        //storing prices of cars in v
	for(i=0;i<n;i++)
	{
	    cin>>v[i];
	}
        //we sort this vector using sort() function
	sort(v.begin(),v.end());
	//calculation of price of cars after depreciation and then adding their sum in k
	for(i=n-1;i>=0;i--)
	{
    	if(v[i]-n+1+i>0)k+=v[i]-n+1+i;
        else break;
	}
	k=k%1000000007;
	//printing maximum possible profit
	cout<<k<<endl;
	return 0;
}
```

**Review by atarax665 — APPROVED** [review]

**Comment by atarax665** [comment]
LGTM 👍

**Comment by shivam-iitk** [comment]
@atarax665 done. Also please check discord dm once.

**Pull request merged**